src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -2312,6 +2312,7 @@ internal bool TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataRead
                             }
                             break;
                         }
+#if !CORE
```

Comment:
I feel like a generic type in the "Core" assembly that the Azure assembly could initialize values on (default null without the application "initializing" them to different values from the Azure assembly version) would be better in these internal types (``TdsParser``, ``SqlInternalConnectionTds``, ``ADP`` (AdapterUtil.cs).
The new generic type also being internal as well so only the azure assembly could use it as well.

For the case of ``TdaParer``, it can store the member information needed to ``Invoke`` the code for that case (with some null checking first to default values of said internal type to null in cases where the azure assembly is not used directly by the application. ``CreateSqlException`` can also fall under this as well, same for the other file where an enum is used to call into Active Directory code (``EnclaveDelegate`` in ``EnclaveDelegate.Crypto.cs``.